### PR TITLE
add logical environment scope to resources without it

### PR DESCRIPTION
**Pull request by reset**
This adds the logical environment scoping to a few resources which were missing it. I will run Terraform on the live environment to bring it inline once this is merged.

gif-keyboard-6671091646682989744

Signed-off-by: Jamie Winsor <jamie@vialstudios.com>

### CODE REVIEW

**Comment by adamhjk** [comment]
gif-keyboard-18237513571405333651

File: terraform/builder-gateways.tf [hunk 1]
```diff
@@ -146,7 +146,7 @@ resource "aws_security_group" "builder_api" {
 }
 
 resource "aws_elb" "builder_api" {
-    name            = "builder-api"
+    name            = "builder-api-${var.env}"
```
Comment (jtimberman) [inline terraform/builder-gateways.tf]: Guess we don't need #998 w/ this ;).

**Comment by reset** [comment]
@thesentinels r+

**Comment by thesentinels** [comment]
:pushpin: Commit f11d20f has been approved by `reset`

**Comment by thesentinels** [comment]
:hourglass: Testing commit f11d20f with merge f11d20f...

**Comment by thesentinels** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/habitat-sh/habitat/builds/140621180)

**Pull request merged**